seaworthy/containers/base.py

```diff
@@ -50,28 +62,82 @@ def __init__(self, name, image, wait_patterns=None, wait_timeout=None,
 
         self._create_kwargs = {} if create_kwargs is None else create_kwargs
 
+        self._docker_helper = docker_helper
         self._container = None
 
-    def create_and_start(self, docker_helper, pull=True, kwargs=None):
+    @property
+    def docker_helper(self):
+        if self._docker_helper is None:
+            raise RuntimeError('No docker_helper set.')
+        return self._docker_helper
+
+    def __enter__(self):
+        self.create_and_start()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._teardown()
+
+    def _teardown(self):
+        """
+        Stop and remove the container if it exists.
+        """
+        if self._container is not None:
+            self.stop_and_remove()
+
+    def as_fixture(self, name=None):
+        if name is None:
+            name = self.name
+
+        def deco(f):
+            @functools.wraps(f)
+            def wrapper(*args, **kw):
+                with self:
+                    kw[name] = self
+                    return f(*args, **kw)
+            return wrapper
+        return deco
+
+    def set_docker_helper(self, docker_helper):
+        if docker_helper is None:  # We don't want to "unset" in this method.
+            return
+        if docker_helper is self._docker_helper:  # We already have this one.
+            return
+        if self._docker_helper is None:
+            self._docker_helper = docker_helper
+        else:
+            raise RuntimeError('Cannot replace existing docker_helper.')
+
+    def status(self):
+        """
+        Get the container's status. If the container does not exist (before
+        creation and after removal), the status is ``None``.
+        """
+        if self._container is None:
+            return None
+        return self.docker_helper.containers.status(self.inner())
+
+    def create_and_start(self, docker_helper=None, pull=True, kwargs=None):
```

Comment:
I'm thinking we should rename the `run`.. as that's basically what it is. But that's a separate PR, probably.